spec/support/features/irs_attempts_api_tracking_helper.rb

```diff
@@ -25,4 +25,9 @@ def mock_irs_attempts_api_encryption_key
   def irs_attempts_api_tracked_events
     IrsAttemptsApiEventDecryptor.new.decrypted_events_from_store.values
   end
+
+  def stub_attempts_tracker
+    controller.irs_attempts_api_tracker = FakeAttemptsTracker.new
+    @irs_attempts_api_tracker = controller.irs_attempts_api_tracker
+  end
```

Comment:
this is great! It will help us simplify so many tests!

I would remove the `attr_writer` and stub instead:

```suggestion
  def stub_attempts_tracker
    irs_attempts_api_tracker = FakeAttemptsTracker.new
    allow(controller).to receive(:irs_attempts_api_tracker).and_return(irs_attempts_api_tracker)
    @irs_attempts_api_tracker = irs_attempts_api_tracker
  end
```

Reply:
Done!  Thanks.